From 83fb2c8551e0264806844f4356c8704119876b0b Mon Sep 17 00:00:00 2001
From: Gabriel Roman <gabrielroman@google.com>
Date: Fri, 15 May 2026 04:18:29 -0700
Subject: [PATCH 2/2] Replace DeviceAdminInfo.loadDescription() with
 loadDescriptionSafe()

The loadDescriptionSafe() truncates the description string to a safe
1024 characters length.

Bug: 492093013
Test: manual
Flag: EXEMPT BUGFIX
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:7087e95c2bed4150f125cda58a1c468873dfe625
Security vulnerability cherry pick for https://b.corp.google.com/issues/492093013
Topic: https://googleplex-android-review.git.corp.google.com/q/topic:%22b492093013%22
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:b94f16d6ac552bff8ab22a413a90c404b1bf8a00
Merged-In: Id34df47391ba7a526810d5d785e140f220e27725
Change-Id: Id34df47391ba7a526810d5d785e140f220e27725
---
 .../settings/enterprise/BaseEnterprisePreferenceController.java | 2 +-
 .../enterprise/DeviceAdminAppsPreferenceController.java         | 2 +-
 .../DeviceAdminAddHeaderPreferenceControllerTest.java           | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/com/android/car/settings/enterprise/BaseEnterprisePreferenceController.java b/src/com/android/car/settings/enterprise/BaseEnterprisePreferenceController.java
index 9f13439..5bed02c 100644
--- a/src/com/android/car/settings/enterprise/BaseEnterprisePreferenceController.java
+++ b/src/com/android/car/settings/enterprise/BaseEnterprisePreferenceController.java
@@ -78,7 +78,7 @@ abstract class BaseEnterprisePreferenceController<P extends Preference>
     @Nullable
     protected final CharSequence getDescription(DeviceAdminInfo deviceAdminInfo) {
         try {
-            return deviceAdminInfo.loadDescription(mPm);
+            return deviceAdminInfo.loadDescriptionSafe(mPm);
         } catch (Resources.NotFoundException e) {
             mLogger.v("No description for "
                     + deviceAdminInfo.getComponent().flattenToShortString());
diff --git a/src/com/android/car/settings/enterprise/DeviceAdminAppsPreferenceController.java b/src/com/android/car/settings/enterprise/DeviceAdminAppsPreferenceController.java
index e4ec3fc..8a45161 100644
--- a/src/com/android/car/settings/enterprise/DeviceAdminAppsPreferenceController.java
+++ b/src/com/android/car/settings/enterprise/DeviceAdminAppsPreferenceController.java
@@ -104,7 +104,7 @@ public abstract class DeviceAdminAppsPreferenceController
         preference.setIcon(deviceAdminInfo.loadIcon(mPm));
         CharSequence description = getDescription(deviceAdminInfo);
         if (!TextUtils.isEmpty(description)) {
-            preference.setSummary(deviceAdminInfo.loadDescription(mPm));
+            preference.setSummary(deviceAdminInfo.loadDescriptionSafe(mPm));
         }
         preference.setKey(deviceAdminInfo.getPackageName());
         ComponentName componentName = deviceAdminInfo.getComponent();
diff --git a/tests/multivalent/src/com/android/car/settings/enterprise/DeviceAdminAddHeaderPreferenceControllerTest.java b/tests/multivalent/src/com/android/car/settings/enterprise/DeviceAdminAddHeaderPreferenceControllerTest.java
index 432a63a..fac91fa 100644
--- a/tests/multivalent/src/com/android/car/settings/enterprise/DeviceAdminAddHeaderPreferenceControllerTest.java
+++ b/tests/multivalent/src/com/android/car/settings/enterprise/DeviceAdminAddHeaderPreferenceControllerTest.java
@@ -178,7 +178,7 @@ public final class DeviceAdminAddHeaderPreferenceControllerTest extends
         verifyPreferenceSetChecked(mPreference, true);
         verifyPreferenceEnabled(mPreference);
         verifyPreferenceTitleSet(mPreference, mFancyDeviceAdminInfo.loadLabel(mRealPm));
-        verifyPreferenceSummarySet(mPreference, mFancyDeviceAdminInfo.loadDescription(mRealPm));
+        verifyPreferenceSummarySet(mPreference, mFancyDeviceAdminInfo.loadDescriptionSafe(mRealPm));
         verifyPreferenceIconSet(mPreference);
     }
 
-- 
2.55.0

